src/pages/basic-concepts/content-markdown.md

```diff
@@ -3,7 +3,143 @@ layout: ~/layouts/Main.astro
 title: Content (Markdown)
 ---
 
-- Layouts
-- MDX
+Astro comes with out-of-the-box Markdown support powered by the expansive [remark](https://remark.js.org/) ecosystem.
 
-## Remote data
+### Remark Plugins
+
+**This is the first draft of Markdown support!** While we plan to support user-provided `remark` plugins soon, our hope is that you won't need `remark` plugins at all!
+
+In addition to [custom components inside the `<Markdown>` component](https://github.com/snowpackjs/astro/blob/main/docs/markdown.md#markdown-component), Astro comes with [Github-flavored Markdown](https://github.github.com/gfm/) support, [Footnotes](https://github.com/remarkjs/remark-footnotes) syntax, [Smartypants](https://github.com/silvenon/remark-smartypants), and syntax highlighting via [Prism](https://prismjs.com/) pre-enabled. These features are likely to be configurable in the future.
+
+### Markdown Pages
+
+Astro treats any `.md` files inside of the `/src/pages` directory as pages. These pages are processed as plain Markdown files and do not support components. If you're looking to embed rich components in your Markdown, take a look at the [Markdown Component](#markdown-component) section.
+
+`layout`
+
+The only special Frontmatter key is `layout`, which defines the relative path to a `.astro` component which should wrap your Markdown content.
+
+`src/pages/index.md`
+
+```jsx
+---
+layout: ../layouts/main.astro
+---
+
+# Hello World!
+```
+
+Layout files are normal `.astro` components. Any Frontmatter defined in your `.md` page will be exposed to the Layout component as the `content` prop. `content` also has an `astro` key which holds special metadata about your file, like the complete Markdown `source` and a `headings` object.
```

Comment:
Could we specify the full shape of the `content` prop that's passed to layouts here?

Something like: https://github.com/snowpackjs/astro/blob/main/docs/api.md#user-content-fetchcontent

```
// ./src/components/my-component.astro
---
const data = Astro.fetchContent('../pages/post/*.md'); // returns an array of posts that live at ./src/pages/post/*.md
---

<div>
{data.slice(0, 3).map((post) => (
  <article>
    <h1>{post.title}</h1>
    <p>{post.description}</p>
    <a href={post.url}>Read more</a>
  </article>
))}
</div>
```

Reply:
That makes sense, I'll make those changes. Thank you!

Reply:
I agree, the content does fit here, maybe we can have a quick overview of that prop shape and how to use it but then defer the user to either a `Reference` or `Data Fetching` post for more defined details of the content prop? Thoughts?

Reply:
Wait I meant it should be
```jsx
{
   /** frontmatter from the post.. example frontmatter:
    title: '',
    tag: '',
    date: '',
    image: '',
    author: '',
    description: '',
   **/
    astro: {
      headers: [], // TODO: document what this means
      source: '' // raw source of the markdown file
    },
    url: '' // the rendered path
  }[]
```

Reply:
maybe `content = { ...` ?

Reply:
I was thinking of just adding a stand alone _Blog Post_ frontmatter example. What do you think? or keep it consistent and simple with just "title" coming from content prop?

ex.
```json
{
  "title": "",
  "date": "",
  "author": "",
  "description": "",
  "astro": {
    "headers": [],
    "source": ""
  },
  "url": ""
}
```

Reply:
Well the reason I had the frontmatter commented out was to signify that it's not guaranteed to be in the `content` object, the only guaranteed variables are `astro` and `url`, no? Or I guess you mean saying

*And here's what an example blog post's `content` variable might look like:*

```
{
  "title": "",
  "date": "",
  "author": "",
  "description": "",
  "astro": {
    "headers": [],
    "source": ""
  },
  "url": ""
}
```

Reply:
Yeah I can explain what you mentioned about it not being guaranteed to be in the content object, etc. But then give that blog example.

Reply:
Sounds good to me!